src/seedu/addressbook/data/person/Block.java

```diff
@@ -0,0 +1,79 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * @author sharkey
+ * Represents a block number in an address
+ * Guarantees immutable; is valid as declared in {@link #isValidName(String)}
+ *
+ */
+public class Block {
+	
+	public static final String EXAMPLE = "950";
+	public static final String MESSAGE_BLOCK_CONSTRAINTS = "A block can be anything";
+	public static final String BLOCK_VALIDATION_REGEX = ".+";
+	public final String block;
```

Comment:
I understand your intent on using public here as the final modifier does not allow block to be changed.  That said, this immutability property is only specific for strings.  In other situations whether the object is not immutable, setting final would only ensure that the reference to the object does not change.  However, one can still access the object via the reference and change the contents of the object.  So in such cases, restricting private access will be in order.